index.bs

```diff
@@ -3927,19 +3933,40 @@ The [=remote end steps=] with |command parameters| are:
    1. Otherwise, set the |context|'s [=layout viewport=] to the
       implementation-defined default.
 
+1. Run the [[cssom-view-1#resizing-viewports]] steps.
+
 1. If |command parameters| [=map/contains=] the <code>devicePixelRatio</code> field:
 
    1. Let |device pixel ratio| be the |command
       parameters|["<code>devicePixelRatio</code>"].
 
-   1. If |device pixel ratio| is not null, change the size of the [=CSS Pixel=]
-      in |context|'s [=layout viewport=] such that it corresponds to |device
-      pixel ratio| in device pixels.
+   1. For the |context| and all [=descendant navigables=]:
 
-   1. Otherwise, set the size of the [=CSS Pixel=] in |context|'s [=layout
-      viewport=] to the implementation-defined default.
+      1. If |device pixel ratio| is not null:
 
-1. Run the [[cssom-view-1#resizing-viewports]] steps.
+         1. When the [=select an image source from a source set=] are run, act as if
+            the implementation's pixel density was set to |device pixel ratio| when selecting an image.
+
+         1. For the purposes of the [=resolution media feature=], act as if
+            the implementation's resolution is |device pixel ratio| dppx scaled by the page zoom.
+
+         1. When the [=determine the device pixel ratio=] steps are run, return |device pixel ratio|.
```

Comment:
Can we at least file an issue to upstream this change? It seems like CSSOM could either call into WebDriver or itself define an override device pixel ratio map that WebDriver would update at this step. I don't like totally monkey patching the algorithm in this case where it's not falling back to implementation-defined behavior.

Reply:
Do we still have to file the issue, since we specified the patch for spec? (sorry, I don't know what the right way to do such things).